### PR TITLE
Added justfile

### PR DESCRIPTION
**Pull request by StevenLove**
[Just](https://github.com/casey/just) is a popular command runner to replace `make`. I have struggled to remember and to type the exact wording of various tests so this allows me to type `just test` and cover all of my bases.  

Also note the script that checks the version of `cargo clippy` in the CI and compares it to your local `cargo clippy` version to remind you to update if necessary

### CODE REVIEW

File: Justfile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     cargo test --package slosh_test --test slosh-tests run_slosh_tests -- --exact --nocapture
+
+# These tests are a subset of 'cargo test --workspace'
+# Test slosh tests defined in /slosh/tests/*.slosh
+test-slosh-tests:
+    cargo test --package slosh --test lisp-scripts -- --nocapture
+
+# Use cspell to spellcheck the codebase
+spellcheck:
+    npx cspell .
```
Comment (gpwclark) [inline Justfile:50]: how do i know i have installed everything I need to run just.

do i install just via cargo?
what is npx?

maybe add something in the README so we know how to approach. I like the idea of a centralized file that I can start sticking useful project stuff in, and the shell script to check the cargo version is quite clever!

Comment (sstanfield) [reply Justfile:50]: I am fine with this, I don't want to required build deps beyond cargo but I don't think this does that.  Just supports another option plus records some handy command.

File: scripts/check-clippy-version.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (sstanfield) [inline scripts/check-clippy-version.sh:1]: Would be nice if we ate our own dog food here.  I know there can be chicken/egg issues but we are building a shell so we should be able to write simple scripts with it.  Or fix it if we can not...

**Review by sstanfield — APPROVED** [review]

File: Justfile (hunk 1 above)
Comment (StevenLove) [reply Justfile:50]: Once you've [installed](https://github.com/casey/just?tab=readme-ov-file#packages) just, you can prove it by running `just -V` in your shell.

npx is another command-line utility, an alteration of npm the node package manager that comes with nodejs. Running `npx cspell .` downloads [cspell](https://www.npmjs.com/package/cspell) into a local cache on your machine and then runs it. And if it already existed in your local cache then of course it doesn't have to download it. This makes it quite simple to make use of nodejs scripts.

File: scripts/check-clippy-version.sh (hunk 2 above)
Comment (StevenLove) [reply scripts/check-clippy-version.sh:1]: Agreed!

File: Justfile (hunk 1 above)
Comment (StevenLove) [reply Justfile:50]: It's also possible to [set up shell completion scripts[(https://github.com/casey/just?tab=readme-ov-file#shell-completion-scripts) which makes it quite quick to see the potential commands

**Review by gpwclark — APPROVED** [review]

**Pull request merged**